src/sign/decentralized_party/signature_threshold_decryption_round.rs

```diff
@@ -156,22 +156,32 @@ where
         Ok((self.nonce_x_coordinate, signature_s))
     }
 
-    /// The lightweight $$ O(1) $$ threshold decryption logic, which simply verifies the output of
-    /// the decryption sent by the designated decrypting party. Blames it in case of an invalid
-    /// signature, and accepts otherwise.
+    /// A wrapper function for [`Self::verify_decrypted_signature_static`] that uses self's
+    /// attributes to verify the decrypted signature.
     pub fn verify_decrypted_signature(
         self,
         signature_s: GroupElement::Scalar,
-        designated_decrypting_party_id: PartyID,
     ) -> crate::Result<(GroupElement::Scalar, GroupElement::Scalar)> {
-        verify_signature(
-            self.nonce_x_coordinate,
+        Self::verify_decrypted_signature_static(
             signature_s,
+            self.nonce_x_coordinate,
             self.message,
             self.public_key,
         )
-        .map_err(|_| Error::MaliciousDesignatedDecryptingParty(designated_decrypting_party_id))?;
+    }
 
-        Ok((self.nonce_x_coordinate, signature_s))
+    /// The lightweight $$ O(1) $$ threshold decryption logic, which simply verifies the output of
+    /// the decryption sent by the designated decrypting party. Returns a [`Error::SignatureVerification`]
+    /// in case of an invalid signature, and accepts otherwise.
+    pub fn verify_decrypted_signature_static(
+        signature_s: GroupElement::Scalar,
+        nonce_x_coordinate: GroupElement::Scalar,
+        message: GroupElement::Scalar,
+        public_key: GroupElement,
+    ) -> crate::Result<(GroupElement::Scalar, GroupElement::Scalar)> {
+        verify_signature(nonce_x_coordinate, signature_s, message, public_key)
+            .map_err(|_| Error::SignatureVerification)?;
```

Comment:
```suggestion
            .map_err(|_| Error::MaliciousDesignatedDecryptingParty)?;
```
Please change it so, and simply remove the `PartyID` from `MaliciousDesignatedDecryptingParty` as I asked before

Reply:
I didn't do it because it's being used [here](https://github.com/dwallet-labs/2pc-mpc/blob/8435c760e7fff5104a58812557544f5fdcf28ac0/src/sign/decentralized_party/identifiable_abort/signature_partial_decryption_verification_round.rs#L116) - do you want me to remove it from there as well?

Reply:
yeah